src/main/java/cuchaz/enigma/network/DedicatedEnigmaServer.java

```diff
@@ -0,0 +1,148 @@
+				.withRequiredArg()
+				.required()
+				.withValuesConvertedBy(Main.PathConverter.INSTANCE);
+
+		OptionSpec<Path> profileOpt = parser.accepts("profile", "Profile json to apply at startup")
+				.withRequiredArg()
+				.withValuesConvertedBy(Main.PathConverter.INSTANCE);
+
+		OptionSpec<Integer> portOpt = parser.accepts("port", "Port to run the server on")
+				.withOptionalArg()
+				.ofType(Integer.class)
+				.defaultsTo(EnigmaServer.DEFAULT_PORT);
+
+		OptionSet parsedArgs = parser.parse(args);
+		Path jar = parsedArgs.valueOf(jarOpt);
+		Path mappingsFile = parsedArgs.valueOf(mappingsOpt);
+		Path profileFile = parsedArgs.valueOf(profileOpt);
+		int port = parsedArgs.valueOf(portOpt);
+
+		System.out.println("Starting Enigma server");
+		DedicatedEnigmaServer server;
+		try {
+			byte[] checksum = Utils.zipSha1(parsedArgs.valueOf(jarOpt));
+
+			EnigmaProfile profile = EnigmaProfile.read(profileFile);
+			Enigma enigma = Enigma.builder().setProfile(profile).build();
+			System.out.println("Indexing Jar...");
+			EnigmaProject project = enigma.openJar(jar, ProgressListener.none());
+
+			MappingFormat mappingFormat = MappingFormat.ENIGMA_DIRECTORY;
+			EntryRemapper mappings;
+			if (!Files.exists(mappingsFile)) {
+				mappings = EntryRemapper.empty(project.getJarIndex());
+			} else {
+				System.out.println("Reading mappings...");
+				if (Files.isDirectory(mappingsFile)) {
+					mappingFormat = MappingFormat.ENIGMA_DIRECTORY;
+				} else if ("zip".equalsIgnoreCase(MoreFiles.getFileExtension(mappingsFile))) {
+					mappingFormat = MappingFormat.ENIGMA_ZIP;
+				} else {
+					mappingFormat = MappingFormat.ENIGMA_FILE;
+				}
+				mappings = EntryRemapper.mapped(project.getJarIndex(), mappingFormat.read(mappingsFile, ProgressListener.none(), profile.getMappingSaveParameters()));
+			}
+
+			PrintWriter log = new PrintWriter(Files.newBufferedWriter(Paths.get("log.txt")));
+
+			server = new DedicatedEnigmaServer(checksum, profile, mappingFormat, mappingsFile, log, mappings, port);
+			server.start();
+			System.out.println("Server started");
+		} catch (IOException | MappingParseException e) {
+			System.err.println("Error starting server!");
+			e.printStackTrace();
+			System.exit(1);
+			return;
+		}
+
+		// noinspection RedundantSuppression
+		// noinspection Convert2MethodRef - javac 8 bug
+		Executors.newScheduledThreadPool(1).scheduleAtFixedRate(() -> server.runOnThread(() -> server.saveMappings()), 0, 1, TimeUnit.MINUTES);
+		Runtime.getRuntime().addShutdownHook(new Thread(server::saveMappings));
+
+		while (true) {
+			try {
+				server.tasks.take().run();
+			} catch (InterruptedException e) {
+				break;
+			}
+		}
+	}
+
+	@Override
+	public synchronized void stop() {
+		super.stop();
+		System.exit(0);
+	}
+
+	private void saveMappings() {
+		mappingFormat.write(getMappings().getObfToDeobf(), getMappings().takeMappingDelta(), mappingsFile, ProgressListener.none(), profile.getMappingSaveParameters());
+		log.flush();
```

Comment:
Maybe add a saving mappings log entry?

Reply:
I thought about that but it could get kind of spammy since it autosaves every minute.